test/logflare_web/plugs/logpush_parser_test.exs

```diff
@@ -0,0 +1,217 @@
+          69, 235, 175, 45, 205, 177, 77, 187, 189, 83, 127, 117, 243, 77, 214, 156, 46, 88, 140,
+          17, 126, 17, 188, 6, 186, 227, 6, 158, 239, 91, 33, 233, 176, 80, 222, 200, 198, 89, 36,
+          74, 142, 69, 145, 77, 140, 243, 113, 148, 83, 85, 67, 79, 76, 32, 22, 50, 233, 237, 37,
+          182, 1, 177, 229, 108, 75, 136, 68, 82, 141, 254, 124, 120, 210, 48, 236, 52, 167, 89,
+          229, 195, 108, 30, 135, 113, 133, 208, 29, 26, 238, 82, 91, 129, 28, 102, 80, 37, 130,
+          97, 19, 232, 167, 128, 244, 194, 125, 97, 126, 251, 221, 219, 185, 214, 133, 58, 26, 12,
+          50, 156, 116, 121, 131, 253, 93, 12, 110, 152, 102, 156, 15, 182, 71, 191, 63, 164, 102,
+          19, 179, 165, 102, 36, 196, 45, 171, 178, 190, 255, 220, 66, 159, 106, 184, 96, 11, 166,
+          171, 246, 60, 10, 146, 209, 159, 5, 58, 84, 15, 97, 142, 59, 235, 193, 74, 155, 93, 76,
+          151, 196, 176, 241, 237, 134, 108, 190, 231, 4, 158, 227, 59, 94, 64, 110, 252, 102, 9,
+          85, 177, 70, 60, 121, 20, 235, 174, 231, 110, 176, 190, 1, 24, 208, 4, 100, 155, 38,
+          246, 173, 41, 37, 249, 137, 202, 135, 105, 212, 220, 179, 29, 195, 50, 137, 17, 134, 6,
+          113, 195, 222, 94, 81, 93, 191, 15, 104, 209, 48, 177, 67, 84, 61, 125, 185, 174, 38,
+          163, 186, 179, 29, 183, 171, 169, 207, 24, 46, 191, 228, 148, 229, 88, 114, 74, 115, 5,
+          221, 6, 189, 4, 165, 104, 6, 15, 51, 107, 203, 184, 4, 61, 23, 21, 179, 198, 195, 89,
+          116, 214, 59, 4, 93, 231, 106, 111, 102, 83, 81, 202, 24, 218, 153, 53, 201, 236, 246,
+          124, 139, 159, 221, 178, 54, 216, 55, 119, 151, 229, 14, 135, 186, 225, 110, 155, 252,
+          249, 155, 251, 127, 166, 88, 87, 215, 87, 163, 103, 244, 41, 147, 64, 117, 63, 19, 84,
+          171, 190, 164, 105, 154, 131, 122, 175, 32, 135, 88, 31, 255, 240, 134, 226, 128, 150,
+          112, 204, 148, 161, 101, 9, 111, 16, 211, 69, 14, 184, 205, 42, 75, 221, 154, 175, 65,
+          191, 68, 135, 224, 61, 234, 123, 166, 251, 95, 233, 87, 67, 41, 219, 114, 2, 43, 13, 44,
+          47, 61, 84, 179, 109, 103, 14, 37, 77, 229, 113, 38, 16, 226, 197, 74, 223, 113, 220,
+          34, 235, 187, 213, 51, 146, 70, 124, 203, 240, 77, 131, 248, 174, 97, 109, 71, 255, 50,
+          125, 123, 70, 220, 136, 23, 120, 193, 19, 226, 246, 117, 8, 219, 11, 218, 245, 183, 194,
+          246, 8, 228, 247, 100, 206, 33, 246, 107, 101, 206, 117, 2, 242, 197, 100, 206, 124,
+          181, 204, 145, 192, 114, 255, 223, 58, 199, 177, 187, 7, 34, 103, 91, 158, 147, 38, 45,
+          35, 159, 22, 57, 226, 133, 158, 255, 47, 68, 206, 176, 108, 215, 32, 150, 105, 56, 175,
+          149, 184, 208, 218, 165, 218, 211, 74, 71, 108, 195, 65, 198, 187, 30, 6, 11, 191, 136,
+          236, 125, 24, 205, 30, 21, 189, 106, 215, 191, 94, 243, 182, 255, 51, 30, 215, 188, 154,
+          57, 175, 82, 189, 191, 0, 29, 159, 90, 32, 167, 13, 0, 0>>
+
+      data = %{
```

Comment:
quite verbose to do a full equality check on the result. Only need to test that the parser is converting, don't need to check full payload.

Can have some match assertions on a few keys and assume the rest will be parsed correctly.


```elixir
assert {:ok, [ %{"batch" => [%{
  "message" => "{\" <> _,
  "metadata" => %{"ClientRequestProtocol"=> "HTTP/1.1"},
  "timestamp" => _
},_] }, %_{}
```

Reply:
should be able to cut this file loc to <50, better for readability.

Reply:
might actually be better to shift this test over to the `LogsControllerTest` to cover whole ingestion for logpush integration.

Maybe can change base to https://github.com/Logflare/logflare/pull/1184 and [extend the cloudflare ingestion tests](https://github.com/Logflare/logflare/pull/1184/files#diff-0acb00fed743ea98b799247a798c5e66ad98bb2d90ef3aa24c0edffcfdf456c4R80) ?

Reply:
We could additionally do this, but there really is no logic for this in the controller so having the plug tested directly I think is a good idea regardless.

Reply:
it'll be fine to test plug directly